draft-ietf-quic-http.md

```diff
@@ -662,17 +679,27 @@ expose implementations to these vulnerabilities.
 
 ## The CONNECT Method {#connect}
 
-The pseudo-method CONNECT (Section 4.3.6 of {{!RFC7231}}) is primarily used with
-HTTP proxies to establish a TLS session with an origin server for the purposes
-of interacting with "https" resources. In HTTP/1.x, CONNECT is used to convert
-an entire HTTP connection into a tunnel to a remote host. In HTTP/2, the CONNECT
-method is used to establish a tunnel over a single HTTP/2 stream to a remote
-host for similar purposes.
+The CONNECT method requests that the recipient establish a tunnel to the
+destination origin server identified by the request-target (Section 4.3.6 of
+{{!RFC7231}}).  It is primarily used with HTTP proxies to establish a TLS
+session with an origin server for the purposes of interacting with "https"
+resources.
 
-A CONNECT request in HTTP/3 functions in the same manner as in HTTP/2. The
-request MUST be formatted as described in Section 8.3 of {{!HTTP2}}. A CONNECT
-request that does not conform to these restrictions is malformed (see
-{{malformed}}). The request stream MUST NOT be closed at the end of the request.
+In HTTP/1.x, CONNECT is used to convert an entire HTTP connection into a tunnel
+to a remote host. In HTTP/2 and HTTP/3, the CONNECT method is used to establish
+a tunnel over a single stream.
+
+A CONNECT request MUST be constructed as follows:
+
+- The ":method" pseudo-header field is set to "CONNECT"
+- The ":scheme" and ":path" pseudo-header fields are omitted
+- The ":authority" pseudo-header field contains the host and port to connect to
+  (equivalent to the authority-form of the request-target of CONNECT requests
+  (see Section 5.3 of [RFC7230]))
+
+A CONNECT request that does not conform to these restrictions is malformed (see
+{{malformed}}).  The request stream MUST NOT be closed at the end of the
+request.
```

Comment:
This last sentence can go.  It is entirely valid for the request stream to be closed without sending anything, though it is admittedly not always very useful.

Reply:
I think it's useful as an exception to the more general requirement that the stream get closed at the end of the request.  But yes, we can reword it to something like "MAY not be closed."  (I would say "need not be closed," except that an exception to a MUST elsewhere feels like it needs its own normative language.)